ImportXml.py

```diff
@@ -0,0 +1,182 @@
+        return res_desc
+
+    Ans = dict()
+    try:
+        f = open(filepath, "r")
+    except OSError:
+        print(f"{filepath} is an invalid xml file")
+        sys.exit()
+
+    soup = BeautifulSoup(f, "xml")
+    secs = soup.find_all("section")
+    res_sections = list()
+    res_questions = list()
+    for sec in secs:
+        sec_id = sec.attrs["id"]
+        sec_title = sec.sectionInfo.contents[3].string
+        for info in sec.find_all("sectionInfo"):
+            sec_text = info.find_all("text")
+            text_p = ""
+            for t in sec_text:
+                text_p += textParse(t.string)
+            sec_info = text_p
+        res_sections.append({"id": sec_id, "title": sec_title, "info": sec_info})
+        ques = sec.find_all("question")
+        for que in ques:
+            que_cat = {}
+            que_sub = {}
+            resps = que.find_all("response")
+            que_que = uniCodeRe(
+                BeautifulSoup(que.find("text").string, "html.parser").get_text()
+            )
+
+            if que.find("directive") is not None:
+                que_descs = que.find("directive").find_all("text")
+                que_desc = concatQuesitonText(que_descs)
+            else:
+                que_desc = ""
+
+            for resp in resps:
+                que_names = {
+                    True: resps[0].attrs["varName"] + " multi answers",
+                    False: resp.attrs["varName"],
+                }
+                que_name = que_names[len(resps) != 1]
+
+                # if under question next level exist "format" means
+                # it is a "longtext" question else "category" question
+                if resp.category is None:
+                    que_type = resp.format.string  # longtext
+                    que_cat = {}
+                    que_sub = {}
+                else:
+                    que_type = "catogory"
+                    que_cat.update(findCategories(resp))
+                    que_sub.update(findSubquestions(que))
+
+            res_questions.append(
+                {
+                    "name": que_name,
+                    "type": que_type,
+                    "question": que_que,
+                    "description": que_desc,
+                    "categories": que_cat,
+                    "sub_questions": que_sub,
+                    "section_id": sec_id,
+                }
+            )
+
+    Ans = {"section": res_sections, "questions": res_questions}
+
+    if writefiles:
+        with open("xml2json.dat", "w") as fw:
+            dump(Ans, fw)
+        with open("sections.dat", "w") as fs:
+            dump(res_sections, fs)
+        with open("quesitons.dat", "w") as fq:
+            dump(res_questions, fq)
+
+    return Ans
+
+
+if __name__ == "__main__":
+    dic = read_questionnaire_structure("./survey_structure.xml", True)
+# with open('write_xml.dat', 'w') as fw:
+#     dump(dic, fw)
+# print(dic)
```

Comment:
I think we should remove this part:

* Comments
* The function is not mean to be used with `if __name__ == "__main__":`

Reply:
ye, sure